src.ts/transaction.ts

```diff
@@ -3,91 +3,199 @@
 import sha256 from 'js-sha256';
 import BN from 'bn.js';
 
-import { Uint128, SendMoneyTransaction, CreateAccountTransaction,
-    SignedTransaction, DeployContractTransaction, FunctionCallTransaction,
-    StakeTransaction, SwapKeyTransaction, AddKeyTransaction,
-    DeleteKeyTransaction, AccessKey,
-    google} from './protos';
-import { base_decode } from './utils/serialize';
-import { Signature } from './utils/key_pair';
+import { base_decode, serialize } from './utils/serialize';
 import { Signer } from './signer';
 
-const TRANSACTION_FIELD_MAP = new Map<Function, string>([
-    [CreateAccountTransaction, 'createAccount'],
-    [DeployContractTransaction, 'deployContract'],
-    [FunctionCallTransaction, 'functionCall'],
-    [SendMoneyTransaction, 'sendMoney'],
-    [StakeTransaction, 'stake'],
-    [SwapKeyTransaction, 'swapKey'],
-    [AddKeyTransaction, 'addKey'],
-    [DeleteKeyTransaction, 'deleteKey'],
-]);
+class Enum {
```

Comment:
it's strange to have Enum under transaction. is there some other name that would work, or at least a comment on what this means?